test/test_http_server.c

```diff
@@ -127,28 +127,90 @@ static int my_post_handler(mtev_http_rest_closure_t *restc, int npats, char **pa
   }
   return EVENTER_READ | EVENTER_WRITE | EVENTER_EXCEPTION;
 }
+static int my_aco_post_handler(mtev_http_rest_closure_t *restc, int npats, char **pats) {
+  (void)npats;
+  (void)pats;
+
+  mtev_http_session_ctx *ctx = restc->http_ctx;
+  mtev_http_request *req = mtev_http_session_request(ctx);
+  struct my_call_closure *cc;
 
+  if (!restc->call_closure) {
+    cc = calloc(1, sizeof *cc);
+    cc->seed = 5381;
+    cc->buffer_size = 1024 * 1024;
+    const char *buffer_size_str = mtev_http_request_querystring(req , "readsize");
+    if(buffer_size_str) {
+      size_t buffer_size;
+      if (sscanf(buffer_size_str, "%zu", &buffer_size) == 1) {
+        mtevL(mtev_error, "Setting buffer size to %zu\n", buffer_size);
+        cc->buffer_size = buffer_size;
+      }
+    }
+    cc->buffer = malloc(cc->buffer_size);
+    cc->delay_us = 0;
+    const char *delay_str = mtev_http_request_querystring(req, "delay");
+    if(delay_str) {
+      cc->delay_us = atof(delay_str) * 1e6;
+      mtevL(mtev_error, "Setting http delay to %f\n", cc->delay_us * 1e-6);
+    }
+    restc->call_closure = cc;
+    restc->call_closure_free = my_cc_free;
+    mtev_http_response_standard(ctx, 200, "OK", "text/plain");
+    (void)mtev_http_response_option_set(ctx, MTEV_HTTP_CLOSE);
+    mtev_http_response_header_set(ctx, "Content-Type", "text/plain");
+  } else {
+    cc = restc->call_closure;
+  }
+
+  int len = 0;
+  do {
+    int mask;
+    len = mtev_http_session_req_consume(ctx, cc->buffer, cc->buffer_size, cc->buffer_size, &mask);
+    if (len > 0) {
+      cc->seed = hash_djb2(cc->seed, cc->buffer, len);
+    }
+  } while(len > 0);
+  /* Avoid busy looping on the socket, spamming the debug logs */
+  usleep(cc->delay_us);
+  mtev_http_response_appendf(ctx, "HASH %" PRId64 "\r\n", cc->seed);
+  mtev_http_response_end(restc->http_ctx);
+  return 0;
```

Comment:
we're not releasing ``cc`` and ``cc->buffer`` because this is a test?

Reply:
cc is attached with a free function to the restc that owns the request itself.  It will be freed when the context is completed.